### PR TITLE
feat(openshift): Support OpenShift 4.x API Groups. Fixes #1631

### LINKED ISSUE #1631: Support OpenShift 4.x
It seems that OpenShift 4 is not supported,  I get the following error when creating a new project:

Code snippet
==========
```
            client.projectrequests().createNew().
                    withNewMetadata().
                    withName(name).
                    endMetadata().
                    done();
```

Error
====

```
io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://api.cluster-01.devcluster.openshift.com:6443/oapi/v1/projectrequests. Received status: Status(apiVersion=v1, code=404, details=null, kind=Status, message=null, metadata=null, reason=null, status=null, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:503)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:442)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:406)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:365)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:234)
	at io.fabric8.openshift.client.dsl.internal.ProjectRequestsOperationImpl.create(ProjectRequestsOperationImpl.java:74)
	at io.fabric8.openshift.client.dsl.internal.ProjectRequestsOperationImpl.lambda$createNew$0(ProjectRequestsOperationImpl.java:89)
	at io.fabric8.openshift.api.model.DoneableProjectRequest.done(DoneableProjectRequest.java:27)
```

### PR DESCRIPTION
**Pull request by gastaldi**
This PR aims to support OpenShift 4.x endpoints Fix #1631

### CODE REVIEW

**Comment by rohanKanojia** [comment]
ok to test

File: kubernetes-client/src/test/resources/test-template.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 {
   "kind": "Template",
-  "apiVersion": "v1",
+  "apiVersion": "template.openshift.io/v1",
```
Comment (gastaldi) [inline kubernetes-client/src/test/resources/test-template.json]: @rohanKanojia I need your expertise here: with this change, all templates must have the `apiVersion` set to `template.openshift.io/v1`, which may break existing functionality when reading template files set to `v1`. Any suggestions or should we let it throw an error indicating that the template needs to be fixed?

Comment (rohanKanojia) [reply kubernetes-client/src/test/resources/test-template.json]: Not sure but we can try adding `notFoundTransformation` entries in `BackwardsCompatibilityInterceptor`. I think this new apiVersion compatible with openshift 3.11 and 3.10 ? I think maybe we can dump `v1` apiVersion users.....

File: openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/ProjectRequestsOperationImpl.java [hunk 2]
```diff
@@ -53,16 +56,24 @@ public boolean isResourceNamespaced() {
 
   @Override
   public URL getRootUrl() {
-    try {
-      return new URL(OpenShiftConfig.wrap(getConfig()).getOpenShiftUrl());
-    } catch (MalformedURLException e) {
-      throw KubernetesClientException.launderThrowable(e);
+    OpenShiftConfig config = OpenShiftConfig.wrap(context.getConfig());
+    OpenShiftClient oc = new DefaultOpenShiftClient(context.getClient(), config);
```
Comment (gastaldi) [inline openshift-client/src/main/java/io/fabric8/openshift/client/dsl/internal/ProjectRequestsOperationImpl.java]: When is this client closed?

**Comment by gastaldi** [comment]
I can confirm that this PR allows kubernetes-client to work in OS4. @rohanKanojia wanna merge it?

**Comment by gastaldi** [comment]
[merge]

**Pull request merged**